src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -213,6 +216,60 @@ internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<byte> byte
             }
         }
 
+        static readonly byte[] _singleZeroByteArray = new byte[1] { 0 };
+
+        internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<ConstantValue> constants, SourceHashAlgorithm hashAlgorithm = SourceHashAlgorithms.Default)
+        {
+            var algorithmName = GetAlgorithmName(hashAlgorithm);
+            using var incrementalHash = IncrementalHash.CreateHash(algorithmName);
+
+            foreach (var constant in constants)
+            {
+                incrementalHash.AppendData(getBytes(constant));
+            }
+
+            return ImmutableArray.Create(incrementalHash.GetHashAndReset());
+
+            static byte[] getBytes(ConstantValue constant)
+            {
+                Debug.Assert(Enum.GetValues(typeof(ConstantValueTypeDiscriminator)).Cast<ConstantValueTypeDiscriminator>().Max()
+                    == ConstantValueTypeDiscriminator.DateTime);
+
+                switch (constant.Discriminator)
+                {
+                    case ConstantValueTypeDiscriminator.Nothing:
+                        return _singleZeroByteArray;
```

Comment:
>return _singleZeroByteArray; [](http://example.com/codeflow?start=24&length=28)

Is this branch reachable?

Reply:
Yes, for null constants. I'll add a comment

Reply:
> for null constants. I'll add a comment

Consider using ```ConstantValueTypeDiscriminator.Null``` instead. "Nothing" is VB term.